keras_hub/src/models/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,610 @@
+    items = zip(level_names, layer_names)
+    outputs = {key: backbone.get_layer(name).output for key, name in items}
+    return Model(inputs=backbone.inputs, outputs=outputs)
+
+
+def extend_branches(inputs, extractor, FPN_depth):
+    """Extends extractor model with a feature pyramid network.
+
+    Args:
+        inputs: tensor, with image input.
+        extractor: Keras Model with level names as outputs.
+        FPN_depth: integer representing the feature pyramid depth.
+
+    Returns:
+        List of extended branch tensors.
+    """
+    features = list(extractor(inputs).values())
+    branches = apply_path_aggregation_FPN(features, FPN_depth, name="pa_fpn")
+    return branches
+
+
+def extend_backbone(backbone, level_names, FPN_depth):
+    """Extends backbone levels with a feature pyramid network.
+
+    Args:
+        backbone: Keras backbone model with the field "pyramid_level_inputs".
+        level_names: list of strings indicating the level names.
+        trainable: boolean indicating if backbone should be optimized.
+        FPN_depth: integer representing the feature pyramid depth.
+
+    Return:
+        Tuple with input image tensor, and list of extended branch tensors.
+    """
+    feature_extractor = build_feature_extractor(backbone, level_names)
+    image = Input(feature_extractor.input_shape[1:])
+    branches = extend_branches(image, feature_extractor, FPN_depth)
+    return image, branches
+
+
+def decode_regression_to_boxes(preds):
+    """Decodes the results of the YOLOV8Detector forward-pass into boxes.
+
+    Returns left / top / right / bottom predictions with respect to anchor
+    points.
+
+    Each coordinate is encoded with 16 predicted values. Those predictions are
+    softmaxed and multiplied by [0..15] to make predictions. The resulting
+    predictions are relative to the stride of an anchor box
+    (and correspondingly relative to the scale of the feature map from which
+    the predictions came).
+    """
+    BOX_REGRESSION_CHANNELS = 64
+    preds_bbox = Reshape((-1, 4, BOX_REGRESSION_CHANNELS // 4))(preds)
+    preds_bbox = ops.nn.softmax(preds_bbox, axis=-1) * ops.arange(
+        BOX_REGRESSION_CHANNELS // 4, dtype="float32"
+    )
+    return ops.sum(preds_bbox, axis=-1)
+
+
+def dist2bbox(distance, anchor_points):
+    """Decodes distance predictions into xyxy boxes.
+
+    Input left / top / right / bottom predictions are transformed into xyxy box
+    predictions based on anchor points.
+
+    The resulting xyxy predictions must be scaled by the stride of their
+    corresponding anchor points to yield an absolute xyxy box.
+    """
+    left_top, right_bottom = ops.split(distance, 2, axis=-1)
+    x1y1 = anchor_points - left_top
+    x2y2 = anchor_points + right_bottom
+    return ops.concatenate((x1y1, x2y2), axis=-1)  # xyxy bbox
+
+
+@keras_hub_export(["keras_hub.models.YOLOV8ObjectDetector"])
+class YOLOV8ObjectDetector(ImageObjectDetector):
```

Comment:
`ModelNameTaskName` and `TaskName` always. So `YOLOV8ImageObjectDetector`. The most common usage will be `ImageObjectDetector.from_preset` which is thankfully shorter.